src/components/dialog/useFocusTrap.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useRef, useCallback } from 'react';
+
+import { focusOnFirstDescendent, focusOnLastDescendent } from './utils';
+
+const FocusRing = ({ topFocusBumperRef, bottomFocusBumperRef, children }) => {
+  return (
+    <>
+      <div ref={topFocusBumperRef} tabIndex="0" />
+      {children}
+      <div ref={bottomFocusBumperRef} tabIndex="0" />
+    </>
+  );
+};
+
+const openDialogs = new Set();
+
+const useFocusTrap = ({ active, sourceRef }) => {
```

Comment:
Is this a common thing to do?

Reply:
This pattern is based on https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html

I've looked into the snippets provided by w3, as well as investigated some other libraries and this is usually what ends up being implemented. It makes sense to me, and it's a whole lot better than leaving the focus behind in the background 🙈

Reply:
Yes it makes sense! I was just a bit surprised by the amount of code required just to be compliant

Reply:
Yup, sadly it's still very hard to do dialogs and popovers right. There's a new `dialog` html element in the works but support is still very sparse: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog